src/backend/crossterm.rs

```diff
@@ -24,36 +23,310 @@ use crate::{
+    /// [alternate screen]: crate::backend#alternate-screen
+    pub fn on_stderr() -> Result<CrosstermBackend<Stderr>, io::Error> {
+        CrosstermBackend::new(io::stderr())
+            .with_raw_mode()?
+            .with_alternate_screen()
+    }
+}
+
+impl CrosstermBackend<Stdout> {
+    /// Creates a new `CrosstermBackend` using [`std::io::stdout()`] as the buffer.
+    ///
+    /// This method also enables [raw mode] and enters the [alternate screen].
+    ///
+    /// # Example
+    ///
+    /// ```rust,no_run
+    /// # use ratatui::backend::{Backend, CrosstermBackend};
+    /// let mut backend = CrosstermBackend::on_stdout()?;
+    /// # std::io::Result::Ok(())
+    /// ```
+    ///
+    /// [raw mode]: crate::backend#raw-mode
+    /// [alternate screen]: crate::backend#alternate-screen
+    pub fn on_stdout() -> Result<CrosstermBackend<Stdout>, io::Error> {
+        CrosstermBackend::new(io::stdout())
+            .with_raw_mode()?
+            .with_alternate_screen()
+    }
+}
+
+impl Default for CrosstermBackend<Stderr> {
+    // impl Default for CrosstermBackend {
+    /// Creates a new `CrosstermBackend` using [`std::io::stderr()`] as the buffer.
+    ///
+    /// It will usually be more convenient to use [`CrosstermBackend::on_stderr()`], which enables
+    /// raw mode and enters the alternate screen.
+    ///
+    /// # Example
+    ///
+    /// ```rust,no_run
+    /// # use std::io::Stderr;
+    /// # use ratatui::backend::CrosstermBackend;
+    /// let mut backend = CrosstermBackend::<Stderr>::default()
+    ///     .with_raw_mode()?
+    ///     .with_alternate_screen()?;
+    /// # std::io::Result::Ok(())
+    /// ```
+    fn default() -> Self {
+        Self::new(io::stderr())
+    }
+}
+
+impl Default for CrosstermBackend<Stdout> {
+    // impl Default for CrosstermBackend {
+    /// Creates a new `CrosstermBackend` using [`std::io::stdout()`] as the buffer.
+    ///
+    /// It will usually be more convenient to use [`CrosstermBackend::on_stdout()`], which enables
+    /// raw mode and enters the alternate screen.
+    ///
+    /// # Example
+    ///
+    /// ```rust,no_run
+    /// # use std::io::Stdout;
+    /// # use ratatui::backend::{Backend, CrosstermBackend};
+    /// let mut backend = CrosstermBackend::<Stdout>::default()
+    ///     .with_raw_mode()?
+    ///     .with_alternate_screen()?;
+    /// # std::io::Result::Ok(())
+    /// ```
+    fn default() -> Self {
+        Self::new(io::stdout())
+    }
+}
+
+impl<W> Drop for CrosstermBackend<W>
+where
+    W: Write,
+{
+    /// Disables raw mode, and mouse capture, leaves the alternate screen and shows the cursor.
+    ///
+    /// # Panics
+    ///
+    /// This method panics if it fails to disable raw mode, leave the alternate screen, disable
+    /// mouse capture, or show the cursor.
+    fn drop(&mut self) {
+        if self.raw_mode_enabled {
+            self.disable_raw_mode().expect("failed to disable raw mode");
```

Comment:
Do you think the failure case here is ever possible for user to see in their terminal? Or is this something that might only happen in tests or CI?

If the former is possible, it'd be nice if we can add a message that says something to the effect of

> Type `reset` and hit enter to fix your terminal state.

Reply:
Good idea. Will do. I'm unsure of this can be triggered in an interactive terminal - likely it is emulator specific?

Reply:
Addressed in latest push - I run all the resets and collect the results.

I went with keeping the panic however as this allows the app to take action on the failure, whereas just logging does not.